src/controller/CHIPDevice.cpp

```diff
@@ -105,6 +105,12 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
     memmove(&serializable.mOpsCreds, &mPairing, sizeof(mPairing));
     serializable.mDeviceId   = Encoding::LittleEndian::HostSwap64(mDeviceId);
     serializable.mDevicePort = Encoding::LittleEndian::HostSwap16(mDevicePort);
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+    serializable.mInterfaceId = 0;
+#else
+    nlSTATIC_ASSERT_PRINT(sizeof(serializable.mInterfaceId) >= sizeof(mInterface), "Size of network interface ID must fit");
+    serializable.mInterfaceId = Encoding::LittleEndian::HostSwap64(mInterface);
```

Comment:
It's still weird to use something defined to do LE to host swapping to do the opposite, as we do here.  And yes, I do know that operation is its own inverse....

Please make sure there's an issue tracking adding a function defined to swap the other way, or updating the docs on HostSwap64, or something, and that it's referenced in a TODO comment here.